services/workflows-service/src/supabase/supabase.service.ts

```diff
@@ -5,38 +5,58 @@ import { ISupabaseService } from './types';
 
 @Injectable()
 export class SupabaseService implements ISupabaseService {
-  private readonly supabaseClient: SupabaseClient;
+  private readonly supabaseClient!: SupabaseClient;
   private readonly logger = new Logger(SupabaseService.name);
 
   constructor(private readonly configService: ConfigService) {
-    const supabaseUrl = this.configService.get<string>('SUPABASE_URL');
+    const telemetryEnabled = this.configService.get('TELEMETRY_ENABLED');
+    const supabaseUrl = this.configService.get<string>('TELEMETRY_SUPABASE_URL');
     const supabaseApiKey = this.configService.get<string>('TELEMETRY_SUPABASE_API_KEY');
-
-    if (!supabaseUrl || !supabaseApiKey) {
-      throw new Error('Supabase URL or API key is missing in configuration');
+    if (telemetryEnabled){
+      if (!supabaseUrl || !supabaseApiKey) {
+        throw new Error('Supabase URL or API key is missing in configuration');
+      }
+      else{
+        this.supabaseClient = createClient(
+          supabaseUrl,
+          supabaseApiKey,
+          {
+            db: { schema: 'public' },
+          }
+        );
+        this.logger.log('Supabase client initialized.');
+        const infradata = require('/tmp/infra.json');
+        this.logInfraData(infradata);
+      }
+    }
+    else {
+      this.logger.log('Telemetry disabled.');
     }
-
-    this.supabaseClient = createClient(supabaseUrl, supabaseApiKey);
-    this.logger.log('Supabase client initialized.');
   }
 
   async logSignIn(url: string): Promise<void> {
-    const { data, error } = await this.supabaseClient.from('logins').insert([{ url }]);
-
-    if (error) {
-      this.logger.error(`Failed to log sign-in: ${error.message}`);
-    } else {
-      this.logger.log(`Sign-in logged successfully: ${data}`);
+    try {
+      const { data, error } = await this.supabaseClient.from('logins').insert([{ url }]);
+      if (error) {
+        this.logger.error(`Failed to log sign-in: ${error.message}`);
+      } else {
+        this.logger.log(`Sign-in logged successfully: ${data}`);
+      }
+    } catch (error: Error | any) {
+      this.logger.error(`Exception to log sign in data: ${error.message}`);
     }
   }
 
   async logInfraData(infradata: JSON): Promise<void> {
-    const { data, error } = await this.supabaseClient.from('infra').insert([infradata]);
-
-    if (error) {
-      this.logger.error(`Failed to log : ${error.message}`);
-    } else {
-      this.logger.log(`Sign-in logged successfully: ${data}`);
+    try {
+      const { data, error } = await this.supabaseClient.from('infra').insert([infradata]);
+      if (error) {
+        this.logger.error(`Failed to log infra data: ${error.message}`);
+      } else {
```

Comment:
for error handling please use SentryModule and notify error

Reply:
do same for other previous function as well

Reply:
Can you give a reference in workflow-service i can refer and close it

Reply:
`sentryService.captureException(error);`

Reply:
tried addressing it in this [commit](https://github.com/ballerine-io/ballerine/pull/2363/commits/cf9ba00e091c948d10c45e9e340ff2f377f3becf) please verify and let me know

Reply:
Please remove SentryService from providers+ exports and use
```
import { SentryModule } from '@/sentry/sentry.module';

imports: [SentryModule],
```
than the service will be injectable in your service